orca/orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/BaseRetrofitExceptionHandler.groovy

```diff
@@ -79,7 +79,7 @@ abstract class BaseRetrofitExceptionHandler implements ExceptionHandler {
             m.declaredAnnotations.findResult { Annotation annotation ->
               annotation
                 .annotationType()
-                .getAnnotation(RestMethod)?.value()
+                .getAnnotation(HTTP)?.value()
```

Comment:
Since I believe retrofit2 responses provide access to the http method, this whole mechanism of searching for annotations can disappear.  Definitely work for a followup PR.

Reply:
Actually, I think we need to revert this change / plan to make this change later as well.  Anyone with custom tasks that may still throw RetrofitError exceptions will get different retry behavior if we make this change now.

Reply:
404 and 400 are no longer retryable when SpinnakerHttpException is generated with retrofit2. So the behaviour will be different if SpinnakerHttpException is generated with RetrofitError.

Reply:
> Do we expect RetrofitErrors even after upgrade?

Not from orca itself, but if people have custom code, then yes.  I think we need to announce in release notes our plan to make this disappear, but give at least a bit of notice before we do.

Reply:
> 404 and 400 are no longer retryable when SpinnakerHttpException is generated with retrofit2. So the behaviour will be different if SpinnakerHttpException is generated with RetrofitError vs retrofit2.

Really?  The [SpinnakerHttpException constructor for retrofit2](https://github.com/spinnaker/spinnaker/blob/b2f481e4dc91368129e3952b0908e50de147875e/kork/kork-retrofit/src/main/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerHttpException.java#L119) makes them not retryable, same as [SpinnakerRetrofitErrorHandler](https://github.com/spinnaker/spinnaker/blob/b2f481e4dc91368129e3952b0908e50de147875e/kork/kork-retrofit/src/main/java/com/netflix/spinnaker/kork/retrofit/exceptions/SpinnakerRetrofitErrorHandler.java#L58).

Reply:
As an example, [in MonitorKatoTaskSpec,](https://github.com/spinnaker/spinnaker/blob/main/orca/orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTaskSpec.groovy#L151) MAX_HTTP_INTERNAL_RETRIES is due to custom generated RetrofitError and SpinnakerHttpException.

Reply:
I think I sent us down the wrong path.  The code here in orca doesn't look at the retryable property.  What's important here is the http method, which is populated in SpinnakerHttpExceptions built with retrofit2, but not with SpinnakerHttpExceptions built with retrofit1/RetrofitError.  So it's the code here that finds the http method by digging for an annotation.

And yes, [MonitorKatoTask](https://github.com/spinnaker/spinnaker/blob/b2f481e4dc91368129e3952b0908e50de147875e/orca/orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/MonitorKatoTask.groovy#L124) is special because it uses RetrySupport, which does look at the retryable property.  I think we're OK there.

Reply:
Yeah, I got the first point that we don't need to dig to get the HTTP method with retrofit2 as we can get it from the exception. Can you confirm if we go ahead and make this change breaking code for potential legacy-RetrofitError-throwers? Or are you suggesting we handle this case in any different way?

Reply:
@jcavanagh or someone from Apple could tell us for sure, or perhaps @jasonmcintosh or @link108 or someone from Harness could, but I think the safe thing is to leave the code that handles RetrofitError/retrofit1 annotations in place since there's almost certainly some plugin or custom code out there depends on it.